### ISSUE TITLE
I really dont know what to do with this.

### ISSUE THREAD

**Issue by toaddoggo** [nodejs/node-gyp#1865]
xit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/swammythecat/vscode/node_modules/native-is-elevated
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.7.0
gyp info using node@10.9.0 | linux | arm64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/vscode/node_modules/native-is-elevated/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/.node-gyp/4.2.9/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/swammythecat/.node-gyp/4.2.9',
gyp info spawn args   '-Dnode_gyp_dir=/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/swammythecat/.node-gyp/4.2.9/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/swammythecat/vscode/node_modules/native-is-elevated',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/swammythecat/vscode/node_modules/native-is-elevated/build'
  CXX(target) Release/obj.target/iselevated/src/iselevated.o
make: g++: Command not found
make: *** [Release/obj.target/iselevated/src/iselevated.o] Error 127
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:239:12)
gyp ERR! System Linux 4.19.44-04937-gd773af751792
iselevated.target.mk:97: recipe for target 'Release/obj.target/iselevated/src/iselevated.o' failed
make: Leaving directory '/home/swammythecat/vscode/node_modules/native-is-elevated/build'
gyp ERR! command "/home/swammythecat/.asdf/installs/nodejs/10.9.0/bin/node" "/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/swammythecat/vscode/node_modules/native-is-elevated

swammythecat@penguin:~/vscode$ yarn && yarn run gulp vscode-linux-arm64-min && yarn run gulp vscode-linux-arm64-build-deb
yarn install v1.17.3
$ node build/npm/preinstall.js
[1/4] Resolving packages...
[2/4] Fetching packages...
info vscode-fsevents@1.2.12: The platform "linux" is incompatible with this module.
info "vscode-fsevents@1.2.12" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.7: The platform "linux" is incompatible with this module.
info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.4: The platform "linux" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
info vscode-windows-ca-certs@0.1.0: The platform "linux" is incompatible with this module.
info "vscode-windows-ca-certs@0.1.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info vscode-windows-registry@1.0.2: The platform "linux" is incompatible with this module.
info "vscode-windows-registry@1.0.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-foreground-love@0.2.0: The platform "linux" is incompatible with this module.
info "windows-foreground-love@0.2.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-mutex@0.3.0: The platform "linux" is incompatible with this module.
info "windows-mutex@0.3.0" is an optional dependency and failed compatibility check. Excluding it from installation.
info windows-process-tree@0.2.4: The platform "linux" is incompatible with this module.
info "windows-process-tree@0.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @microsoft/applicationinsights-web@2.1.1" has unmet peer dependency "tslib@^1.9.3".
[4/4] Building fresh packages...
[1/13] ⠄ keytar
[2/13] ⠄ native-is-elevated
[3/13] ⠄ native-keymap
[4/13] ⠄ native-watchdog
error /home/swammythecat/vscode/node_modules/native-is-elevated: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/swammythecat/vscode/node_modules/native-is-elevated
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.7.0
gyp info using node@10.9.0 | linux | arm64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/vscode/node_modules/native-is-elevated/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/swammythecat/.node-gyp/4.2.9/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/swammythecat/.node-gyp/4.2.9',
gyp info spawn args   '-Dnode_gyp_dir=/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/home/swammythecat/.node-gyp/4.2.9/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/swammythecat/vscode/node_modules/native-is-elevated',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/home/swammythecat/vscode/node_modules/native-is-elevated/build'
  CXX(target) Release/obj.target/iselevated/src/iselevated.o
make: g++: Command not found
make: *** [Release/obj.target/iselevated/src/iselevated.o] Error 127
gyp ERR! build erroriselevated.target.mk:97: recipe for target 'Release/obj.target/iselevated/src/iselevated.o' failed
make: Leaving directory '/home/swammythecat/vscode/node_modules/native-is-elevated/build'
 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:239:12)
gyp ERR! System Linux 4.19.44-04937-gd773af751792
gyp ERR! command "/home/swammythecat/.asdf/installs/nodejs/10.9.0/bin/node" "/home/swammythecat/.asdf/installs/nodejs/10.9.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/swammythecat/vscode/node_modules/native-is-elevated

swammythecat@penguin:~/vscode$

**Comment by rvagg** [comment]
@toaddoggo the key line is:
>  make: g++: Command not found
You need a C++ compiler installed. By the look of your path names you're on Linux, in which case you'll need to install them from your appropriate package manager.
If you're on a .deb based system (Ubuntu, Debian, etc), try `sudo apt-get install gcc g++ make`. If you're on an .rpm based system (CentOS, RHEL, Fedora) try `sudo yum install gcc-c++ make` or `sudo dnf install gcc-c++ make`.

**Comment by cclauss** [comment]
@rvagg Python’s builtin [platform](https://docs.python.org/3.7/library/platform.html) package allows us to print a lot of stats about the current machine.  Would it help your debugging efforts to have more printed in configure?
https://travis-ci.com/cclauss/node-configure/jobs/228265869#L187

**Comment by rvagg** [comment]
@cclauss maybe .. and maybe we should build some basic reporting that suggests next steps to the user.
I was surprised just now to realise we didn't even list basic Linux package manager instructions on https://github.com/nodejs/node-gyp#on-unix, which we really should. When I did the docs for https://github.com/nodesource/distributions I made sure to make it clear and easy, we need to do the same here.
